flang/include/flang/Optimizer/CodeGen/CodeGen.h

```diff
@@ -35,9 +35,24 @@ struct TargetRewriteOptions {
 std::unique_ptr<mlir::OperationPass<mlir::ModuleOp>> createFirTargetRewritePass(
     const TargetRewriteOptions &options = TargetRewriteOptions());
 
-/// Convert FIR to the LLVM IR dialect
+/// FIR to LLVM translation pass options.
+struct FIRToLLVMPassOptions {
+  // Do not fail when type descriptors are not found when translating
+  // operations that uses them at the LLVM level like fir.embox. Instead,
+  // just use a null pointer.
+  // This is useful to test translating programs manually written where a
+  // frontend did not generate type descriptor data structures. However, note
+  // that this programs would crash at runtime if the derived type descriptors
+  // are required by the runtime, so this only an option to help debugging.
+  bool ignoreMissingTypeDescriptors = false;
```

Comment:
FWIW, FIR originally defined an Op to create a type descriptor, `fir.gentypedesc <type>`. This wasn't fully implemented, but it was intended to allow code gen to lazily create type descriptor information as needed.

That may be helpful to (1) discard type descriptor cruft that isn't really needed at compile-time, (2) create dummy type descriptors for testing that won't necessarily crash (like a nullptr), etc.